csi/controller_server.go

```diff
@@ -270,29 +270,26 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 
 func (cs *ControllerServer) getBackupVolume(volumeName string) (*longhornclient.BackupVolume, error) {
 	log := cs.log.WithFields(logrus.Fields{"function": "getBackupVolume"})
-	vol, err := cs.apiClient.Volume.ById(volumeName)
+	list, err := cs.apiClient.BackupVolume.List(&longhornclient.ListOpts{})
 	if err != nil {
-		return nil, errors.Wrapf(err, "getBackupVolume: fail to get source volume %v", volumeName)
+		return nil, err
+	}
+	if len(list.Data) == 0 {
+		return nil, nil
 	}
 
-	list, err := cs.apiClient.BackupVolume.List(&longhornclient.ListOpts{
-		Filters: map[string]interface{}{
-			types.LonghornLabelBackupTarget: vol.BackupTargetName,
-			types.LonghornLabelBackupVolume: volumeName,
-		}})
+	vol, err := cs.apiClient.Volume.ById(volumeName)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to list backup volumes")
+		return nil, errors.Wrapf(err, "getBackupVolume: fail to get source volume %v", volumeName)
 	}
 
-	if len(list.Data) > 1 {
-		return nil, fmt.Errorf("found multiple backup volumes for backup target %s and volume %s", vol.BackupTargetName, volumeName)
-	}
-	if len(list.Data) == 0 {
-		log.Debugf("cannot find backup volume with backup target %s and volume %s", vol.BackupTargetName, volumeName)
-		return nil, nil
+	for _, bv := range list.Data {
+		if bv.BackupTargetName == vol.BackupTargetName && bv.VolumeName == volumeName {
+			return &bv, nil
+		}
 	}
-
-	return &list.Data[0], nil
+	log.Debugf("cannot find backup volume with backup target %s and volume %s", vol.BackupTargetName, volumeName)
+	return nil, nil
```

Comment:
Why not `return nil, fmt.Errorf(....)"

Reply:
As we discussed before, we return the error as nil when we can not find a backup volume as the backup is not completed. https://github.com/longhorn/longhorn-manager/blob/master/csi/controller_server.go#L1247-L1258

Reply:
Oh, I got it. Can you add a comment here.

Reply:
Added. Thanks